src/hotspot/os/linux/os_linux.cpp

```diff
@@ -216,18 +216,18 @@ julong os::Linux::available_memory() {
 }
 
 julong os::physical_memory() {
-  jlong phys_mem = 0;
+  jlong phys_mem = Linux::physical_memory();
+
   if (OSContainer::is_containerized()) {
     jlong mem_limit;
-    if ((mem_limit = OSContainer::memory_limit_in_bytes()) > 0) {
+    if ((mem_limit = OSContainer::memory_limit_in_bytes()) > 0 && mem_limit < phys_mem) {
       log_trace(os)("total container memory: " JLONG_FORMAT, mem_limit);
       return mem_limit;
     }
     log_debug(os, container)("container memory limit %s: " JLONG_FORMAT ", using host value",
                             mem_limit == OSCONTAINER_ERROR ? "failed" : "unlimited", mem_limit);
```

Comment:
These log lines now result in messages like:

```
[0.037s][debug][os,container] container memory limit unlimited: 38654705664, using host value
```

It looks a bit strange, so perhaps we should account for the exceeding-host-memory case. Thoughts?

Reply:
You're right. I've hopefully addressed this in  `8d7e80c6253`